core/src/test/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManagerSuite.scala

```diff
@@ -30,86 +46,15 @@ class HadoopDelegationTokenManagerSuite extends SparkFunSuite {
     val manager = new HadoopDelegationTokenManager(new SparkConf(false), hadoopConf, null)
     assert(manager.isProviderLoaded("hadoopfs"))
     assert(manager.isProviderLoaded("hbase"))
-    assert(manager.isProviderLoaded("hive"))
-    assert(manager.isProviderLoaded("kafka"))
-  }
-
-  test("disable hive credential provider") {
-    val sparkConf = new SparkConf(false).set("spark.security.credentials.hive.enabled", "false")
-    val manager = new HadoopDelegationTokenManager(sparkConf, hadoopConf, null)
-    assert(manager.isProviderLoaded("hadoopfs"))
-    assert(manager.isProviderLoaded("hbase"))
-    assert(!manager.isProviderLoaded("hive"))
-    assert(manager.isProviderLoaded("kafka"))
+    // This checks that providers are loaded independently and they have no effect on each other
+    assert(!manager.isProviderLoaded("throw"))
```

Comment:
For paranoia, can you make sure the constructor was actually called? (e.g. by updating some variable in an `object` or something)

Reply:
Yep, that's a really important feature not to kill the whole security. Adding...